integration-tests/src/tests/client/features/yield_timeouts.rs

```diff
@@ -0,0 +1,366 @@
+    assert_eq!(env.clients[0].process_tx(tx, false, false), ProcessTxResponse::ValidTx);
+
+    // Allow two blocks for the contract to be deployed
+    for i in 1..3 {
+        env.produce_block(0, i);
+    }
+    assert!(matches!(
+        env.clients[0].chain.get_final_transaction_result(&tx_hash).unwrap().status,
+        FinalExecutionStatus::SuccessValue(_),
+    ));
+
+    // Submit transaction making a function call which will invoke yield create
+    let yield_transaction = SignedTransaction::from_actions(
+        10,
+        "test0".parse().unwrap(),
+        "test0".parse().unwrap(),
+        &signer,
+        vec![Action::FunctionCall(Box::new(FunctionCallAction {
+            method_name: "call_yield_create_return_promise".to_string(),
+            args: anticipated_yield_payload,
+            gas: 300_000_000_000_000,
+            deposit: 0,
+        }))],
+        *genesis_block.hash(),
+    );
+    let yield_tx_hash = yield_transaction.get_hash();
+    assert_eq!(
+        env.clients[0].process_tx(yield_transaction, false, false),
+        ProcessTxResponse::ValidTx
+    );
+
+    // Allow two blocks for the function call to be executed
+    for i in 3..5 {
+        env.produce_block(0, i);
+    }
+
+    let yield_data_ids = find_yield_data_ids_from_latest_block(&env);
+    assert_eq!(yield_data_ids.len(), 1);
+
+    (env, yield_tx_hash, yield_data_ids[0])
+}
+
+/// Add a transaction which invokes yield resume using given data id.
+fn invoke_yield_resume(
+    env: &mut TestEnv,
+    data_id: CryptoHash,
+    yield_payload: Vec<u8>,
+) -> CryptoHash {
+    let signer = InMemorySigner::from_seed("test0".parse().unwrap(), KeyType::ED25519, "test0");
+    let genesis_block = env.clients[0].chain.get_block_by_height(0).unwrap();
+
+    let resume_transaction = SignedTransaction::from_actions(
+        200,
+        "test0".parse().unwrap(),
+        "test0".parse().unwrap(),
+        &signer,
+        vec![Action::FunctionCall(Box::new(FunctionCallAction {
+            method_name: "call_yield_resume".to_string(),
+            args: yield_payload.into_iter().chain(data_id.as_bytes().iter().cloned()).collect(),
+            gas: 300_000_000_000_000,
+            deposit: 0,
+        }))],
+        *genesis_block.hash(),
+    );
+    let tx_hash = resume_transaction.get_hash();
+    assert_eq!(
+        env.clients[0].process_tx(resume_transaction, false, false),
+        ProcessTxResponse::ValidTx
+    );
+    tx_hash
+}
+
+/// Add a bunch of function call transactions, congesting the chain.
+///
+/// Note that these transactions start to be processed in the *second* block produced after they are
+/// inserted to client 0's mempool.
+fn create_congestion(env: &mut TestEnv) {
+    let signer = InMemorySigner::from_seed("test0".parse().unwrap(), KeyType::ED25519, "test0");
+    let genesis_block = env.clients[0].chain.get_block_by_height(0).unwrap();
+
+    let mut tx_hashes = vec![];
+
+    for i in 0..25 {
```

Comment:
Is there not a good way to avoid hardcoding the number of transactions here? I feel like this test might lose its behaviour if the compute costs are reduced in the future.

Reply:
Unfortunately I couldn't think of a way to congest the chain which doesn't involve specifying some fixed number of transactions greater than 1.